# CCTV cameras in countries and major cities in the world

The article ['The world's most-surveilled cities - Comparitech'](https://www.comparitech.com/vpn-privacy/the-worlds-most-surveilled-cities/) states that there is a very low correlation between number of security camera in major cities and crime rate. Let's use the data set given in the article to reproduce the result for the cities in the list below.

 - [Preparing the data](#Preparing-the-data)

## Preparing the data

In [1]:
import numpy as np
import pandas as pd
import os
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [2]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_colwidth', -1)

In [3]:
ROOT_DIR = '/Users/angelo/Programming/data/CCTVs'
cctv_per_city_file_name = os.path.join(ROOT_DIR, 'cctv-per-city.csv')
cctv_per_city_df = pd.read_csv(cctv_per_city_file_name, usecols=range(0, 9), thousands=',')

In [4]:
cctv_per_city_df.head()

City Country  # of CCTV Cameras  # of People   \
0  Chongqing  China   2579890.0          15354067       
1  Shenzhen   China   1929600.0          12128721       
2  Shanghai   China   2985984.0          26317104       
3  Tianjin    China   1244160.0          13396402       
4  Ji'nan     China   540463.0           7321200        

   # of CCTV Cameras per 1,000 People  \
0  168.03                               
1  159.09                               
2  113.46                               
3  92.87                                
4  73.82                                

   Future Projections (or Addition of Private Cameras)  \
0  8075056.0                                             
1  16680000.0                                            
2  9346129.0                                             
3  3894220.0                                             
4  1691649.0                                             

   Future Projections of # of CCTV Cameras per 1,000 People (or Total Inc. Private Cameras)  \
0  525.92                                                                                     
1  1375.25                                                                                    
2  355.14                                                                                     
3  290.69                                                                                     
4  231.06                                                                                     

   Crime Index  Safety Index  
0  33.18        66.82         
1  42.91        57.09         
2  40.87        59.13         
3  29.15        70.85         
4  15.93        84.07

In [5]:
cctv_per_city_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 9 columns):
City                                                                                        120 non-null object
Country                                                                                     120 non-null object
# of CCTV Cameras                                                                           117 non-null float64
# of People                                                                                 120 non-null int64
# of CCTV Cameras per 1,000 People                                                          119 non-null float64
Future Projections (or Addition of Private Cameras)                                         58 non-null float64
Future Projections of # of CCTV Cameras per 1,000 People (or Total Inc. Private Cameras)    58 non-null float64
Crime Index                                                                                 120 non-null float6

In [6]:
cctv_per_city_df.describe().applymap('{:.2f}'.format)

# of CCTV Cameras # of People  # of CCTV Cameras per 1,000 People  \
count  117.00            120.00       119.00                              
mean   115480.84         6450933.74   9.55                                
std    429959.54         6487371.75   26.82                               
min    40.00             128793.00    0.00                                
25%    1200.00           1568572.00   0.40                                
50%    4000.00           4146242.50   1.07                                
75%    24500.00          9746047.75   5.31                                
max    2985984.00        37435191.00  168.03                              

      Future Projections (or Addition of Private Cameras)  \
count  58.00                                                
mean   914311.24                                            
std    2760438.67                                           
min    80.00                                                
25%    8506.00                                              
50%    50056.50                                             
75%    157439.00                                            
max    16680000.00                                          

      Future Projections of # of CCTV Cameras per 1,000 People (or Total Inc. Private Cameras)  \
count  58.00                                                                                     
mean   78.38                                                                                     
std    211.52                                                                                    
min    0.10                                                                                      
25%    2.09                                                                                      
50%    8.03                                                                                      
75%    33.30                                                                                     
max    1375.25                                                                                   

      Crime Index Safety Index  
count  120.00      120.00       
mean   44.95       55.05        
std    16.73       16.73        
min    8.09        19.60        
25%    32.89       42.76        
50%    43.14       56.86        
75%    57.24       67.11        
max    80.40       91.91

Let's do some grouping to get a better sense for the data

In [7]:
cctv_per_city_df.groupby('Country')[['# of CCTV Cameras']].sum().sort_values(by='# of CCTV Cameras', ascending=False)[:10].reset_index()

Country  # of CCTV Cameras
0  China          11407891.0       
1  England (UK)   627707.0         
2  India          274784.0         
3  Russia         176000.0         
4  Iraq           120000.0         
5  Turkey         107000.0         
6  Singapore      86000.0          
7  United States  69038.0          
8  Australia      62565.0          
9  UAE            55000.0

China is the country that has the most surveillance cameras per 1,000 people in the world, followed by England.

In [11]:
cctv_per_city_df.groupby('Country')[['# of CCTV Cameras per 1,000 People']].mean().sort_values(by='# of CCTV Cameras per 1,000 People', ascending=False)[:10].reset_index()

Country  # of CCTV Cameras per 1,000 People
0  China         71.670                            
1  England (UK)  68.400                            
2  Singapore     15.250                            
3  UAE           12.955                            
4  Iraq          12.300                            
5  Germany       11.180                            
6  Russia        8.615                             
7  Poland        7.850                             
8  Austria       7.380                             
9  Turkey        7.150

In [9]:
cctv_per_city_df[['City', '# of CCTV Cameras per 1,000 People', 'Country']].sort_values(by='# of CCTV Cameras per 1,000 People', ascending=False)[:10]

City  # of CCTV Cameras per 1,000 People        Country
0  Chongqing  168.03                              China        
1  Shenzhen   159.09                              China        
2  Shanghai   113.46                              China        
3  Tianjin    92.87                               China        
4  Ji'nan     73.82                               China        
5  London     68.40                               England (UK) 
6  Wuhan      60.49                               China        
7  Guangzhou  52.75                               China        
8  Beijing    39.93                               China        
9  Atlanta    15.56                               United States

Of the top 10 most-surveilled cities in the world, 8 are in China!

In [17]:
cctv_per_city_df.loc[cctv_per_city_df['Country'] == 'England (UK)']

City       Country  # of CCTV Cameras  # of People   \
5  London  England (UK)  627707.0           9176530        

   # of CCTV Cameras per 1,000 People  \
5  68.4                                 

   Future Projections (or Addition of Private Cameras)  \
5 NaN                                                    

   Future Projections of # of CCTV Cameras per 1,000 People (or Total Inc. Private Cameras)  \
5 NaN                                                                                         

   Crime Index  Safety Index  
5  52.24        47.76

In [22]:
cctv_per_city_df.loc[cctv_per_city_df['Country'] == 'China'].reset_index()

index       City Country  # of CCTV Cameras  # of People   \
0   0      Chongqing  China   2579890.0          15354067       
1   1      Shenzhen   China   1929600.0          12128721       
2   2      Shanghai   China   2985984.0          26317104       
3   3      Tianjin    China   1244160.0          13396402       
4   4      Ji'nan     China   540463.0           7321200        
5   6      Wuhan      China   500000.0           8266273        
6   7      Guangzhou  China   684000.0           12967862       
7   8      Beijing    China   800000.0           20035455       
8   13     Urumqi     China   43394.0            3500000        
9   20     Nanchang   China   50400.0            5463538        
10  26     Changsha   China   50000.0            7918100        

    # of CCTV Cameras per 1,000 People  \
0   168.03                               
1   159.09                               
2   113.46                               
3   92.87                                
4   73.82                                
5   60.49                                
6   52.75                                
7   39.93                                
8   12.40                                
9   9.22                                 
10  6.31                                 

    Future Projections (or Addition of Private Cameras)  \
0   8075056.0                                             
1   16680000.0                                            
2   9346129.0                                             
3   3894220.0                                             
4   1691649.0                                             
5   4695000.0                                             
6   2140920.0                                             
7   2504000.0                                             
8   135823.0                                              
9   157752.0                                              
10  156500.0                                              

    Future Projections of # of CCTV Cameras per 1,000 People (or Total Inc. Private Cameras)  \
0   525.92                                                                                     
1   1375.25                                                                                    
2   355.14                                                                                     
3   290.69                                                                                     
4   231.06                                                                                     
5   567.97                                                                                     
6   165.09                                                                                     
7   124.98                                                                                     
8   38.81                                                                                      
9   28.87                                                                                      
10  19.76                                                                                      

    Crime Index  Safety Index  
0   33.18        66.82         
1   42.91        57.09         
2   40.87        59.13         
3   29.15        70.85         
4   15.93        84.07         
5   21.18        78.82         
6   47.43        52.57         
7   42.31        57.69         
8   36.76        63.24         
9   8.09         91.91         
10  30.37        69.63

In [30]:
cctv_per_city_df.groupby('Country')[['City']].count().sort_values(by='City', ascending=False)[:20]

City
Country               
United States     13  
China             11  
Brazil            10  
India             9   
Australia         3   
Pakistan          3   
Japan             2   
France            2   
South Korea       2   
Italy             2   
South Africa      2   
Colombia          2   
UAE               2   
Canada            2   
Mexico            2   
Vietnam           2   
Russia            2   
Northern Ireland  1   
Romania           1   
Norway            1

## Looking for correlation

In [38]:
cctv_per_city_df.corr()

# of CCTV Cameras  \
# of CCTV Cameras                                                                         1.000000            
# of People                                                                               0.381571            
# of CCTV Cameras per 1,000 People                                                        0.933100            
Future Projections (or Addition of Private Cameras)                                       0.873851            
Future Projections of # of CCTV Cameras per 1,000 People (or Total Inc. Private Cameras)  0.729861            
Crime Index                                                                              -0.115621            
Safety Index                                                                              0.115618            

                                                                                          # of People   \
# of CCTV Cameras                                                                         0.381571       
# of People                                                                               1.000000       
# of CCTV Cameras per 1,000 People                                                        0.298218       
Future Projections (or Addition of Private Cameras)                                       0.296488       
Future Projections of # of CCTV Cameras per 1,000 People (or Total Inc. Private Cameras)  0.164595       
Crime Index                                                                               0.110286       
Safety Index                                                                             -0.110297       

                                                                                          # of CCTV Cameras per 1,000 People  \
# of CCTV Cameras                                                                         0.933100                             
# of People                                                                               0.298218                             
# of CCTV Cameras per 1,000 People                                                        1.000000                             
Future Projections (or Addition of Private Cameras)                                       0.913443                             
Future Projections of # of CCTV Cameras per 1,000 People (or Total Inc. Private Cameras)  0.844332                             
Crime Index                                                                              -0.168128                             
Safety Index                                                                              0.168126                             

                                                                                          Future Projections (or Addition of Private Cameras)  \
# of CCTV Cameras                                                                         0.873851                                              
# of People                                                                               0.296488                                              
# of CCTV Cameras per 1,000 People                                                        0.913443                                              
Future Projections (or Addition of Private Cameras)                                       1.000000                                              
Future Projections of # of CCTV Cameras per 1,000 People (or Total Inc. Private Cameras)  0.944398                                              
Crime Index                                                                              -0.194285                                              
Safety Index                                                                              0.194285                                              

                                                                                          Future Projections of # of CCTV Cameras per 1,000 People (or Total Inc. Priv

There is only a -0.168 correlation between number of CCTV cameras per one thousand people and the crime index.

In [10]:
# cctv_per_country_file_name = os.path.join(ROOT_DIR, 'cctv-per-country.csv')
# cctv_per_country_df = pd.read_csv(cctv_per_country_file_name, index_col=0, usecols=range(0, 9))